### PR TITLE
Add a CALIOPEN tag on inbound message that are from an internal domain sender

### PR DESCRIPTION
**Pull request by gdchamal**
A first tag CALIOPEN will appear automatically on incoming message from a sender defined in configuration file (hosted domains).

This should help validate UX/UI around tag display

### CODE REVIEW

**Comment by LaurentChemla** [comment]
\o/

File: src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py [hunk 1]
```diff
@@ -44,7 +48,20 @@ def __init__(self, user):
...
         """Evaluate user rules to get all tags for a mail."""
-        return []
+        tags = []
+        if message.privacy_features.get('is_internal', False):
+            # XXX do not hardcode the wanted tag
+            internal_tag = [x for x in self.user.tags if x.name == 'CALIOPEN']
+            if internal_tag:
+                # XXX need to transform UserTag to ResourceTag
+                tag = ResourceTag()
+                tag.date_insert = datetime.utcnow()
```
Comment (sapiens-sapide) [inline src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py]: I'm not on the same wavelength… `date_insert` is the date the tag was created/inserted in db, not the date it was nested into the object.
Furthermore, we'll change the model: `tags` in messages, contacts, etc. will be an array of IDs, not an object.

Comment (gdchamal) [reply src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py]: Ok I'll wait for your update on nested tag structure to go further.

**Comment by sapiens-sapide** [comment]
Update : new tag API must have a **lowercase** tag.name, not `CALIOPEN`

**Comment by gdchamal** [comment]
[Auto tag caliopen internal messages](https://trello.com/c/1kYMvVSM/7-auto-tag-caliopen-internal-messages)

**Comment by sapiens-sapide** [comment]
« caliopen » is not a generic name for this kind of system label. Should be « internal » or something else that can be use by other instances than Caliopen ones.  
There is the Tag's `label` property to put « caliopen ».

**Pull request merged**